packages/embark-ui/src/containers/BlocksContainer.js

```diff
@@ -28,56 +28,64 @@ class BlocksContainer extends Component {
     this.props.stopBlockHeader();
   }
 
+  get numberOfBlocks() {
+    const blocks = this.props.blocks;
+    return !blocks.length ? 0 : blocks[0].number + 1;
+  }
+
   getNumberOfPages() {
-    if (!this.numberOfBlocks) {
-      let blocks = this.props.blocks;
-      if (blocks.length === 0) {
-        this.numberOfBlocks = 0;
-      } else {
-        this.numberOfBlocks = blocks[blocks.length - 1].number - 1;
-      }
-    }
-    return Math.ceil(this.numberOfBlocks / MAX_BLOCKS);
+    return Math.ceil(this.numberOfBlocks / this.numBlocksToDisplay);
   }
 
   changePage(newPage) {
     this.setState({currentPage: newPage});
-
-    this.props.fetchBlocks((newPage * MAX_BLOCKS) + MAX_BLOCKS);
+    this.props.fetchBlocks(
+      this.numberOfBlocks - 1 - (this.numBlocksToDisplay * (newPage - 1))
+    );
   }
 
   getCurrentBlocks() {
-    const currentPage = this.state.currentPage || this.getNumberOfPages();
-    return this.props.blocks.filter(block => block.number <= (currentPage * MAX_BLOCKS) + MAX_BLOCKS &&
-      block.number > currentPage * MAX_BLOCKS);
+    return this.props.blocks
+      .filter(block => {
+        const index = (
+          (this.numberOfBlocks -
+           (this.numBlocksToDisplay * (this.state.currentPage - 1))) -
+            block.number
+        );
+        return index <= this.numBlocksToDisplay && index > 0;
+      });
   }
 
   render() {
     const newBlocks = this.getCurrentBlocks();
-    if (newBlocks.length) {
-      this.currentBlocks = newBlocks;
-    }
     return (
       <React.Fragment>
-        <PageHead title="Blocks" enabled={this.props.overridePageHead} description="Summary of the most recent blocks" />
-        <DataWrapper shouldRender={this.currentBlocks.length > 0} {...this.props} render={() => (
-          <Blocks blocks={this.currentBlocks} numberOfPages={this.getNumberOfPages()}
+        <PageHead title="Blocks"
+                  enabled={this.props.overridePageHead}
+                  description="Summary of the most recent blocks" />
+        <DataWrapper shouldRender={true} {...this.props} render={() => (
```

Comment:
Could be a separate PR, but don't forget that if we are always rendering the `Blocks` component, we should add a message like "No blocks" when `!this.props.blocks.length`.

Reply:
Done. See: https://github.com/embark-framework/embark/pull/1427/files#diff-82b2992c782c672b1a5ac051f580fdd4R17.